rust-src/bulletproofs/src/set_non_membership_proof.rs

```diff
@@ -0,0 +1,674 @@
+//! Implementation of set-non-membership proof along the lines of bulletproofs
+use crate::{inner_product_proof::*, utils::*};
+use crypto_common::*;
+use crypto_common_derive::*;
+use curve_arithmetic::{multiexp, multiexp_table, multiexp_worker_given_table, Curve};
+use ff::Field;
+use pedersen_scheme::*;
+use rand::*;
+use random_oracle::RandomOracle;
+use std::iter::once;
+
+/// Bulletproof style set-non-membership proof
+#[derive(Clone, Serialize, SerdeBase16Serialize, Debug)]
+#[allow(non_snake_case)]
+pub struct SetNonMembershipProof<C: Curve> {
+    /// Commitment to the multiplicative inverse of `v-s_i` for each `i`
```

Comment:
I would call A a commitment to the multiplicative inverse of `v-s_i` for each `i`, and an `all-v` vector

Reply:
Indeed! The same mistake was there for the set-membership and range-proofs as well. I've updated all of the comments. Please check.

Reply:
I checked and it looks fine.